### PR TITLE
Feature/reassign

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Basic reassign functionality

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Checklist:

-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/apps/ScopeChangeRequest/Components/Workflow/WorkflowCriterias.tsx [hunk 1]
```diff
@@ -1,55 +1,137 @@
...
-                                    >{`${day}/${month}/${year} ${hour}:${paddedMinutes} - ${criteria.signedBy.firstName} ${criteria.signedBy.lastName} `}</div>
-                                ) : (
-                                    <div style={{ fontSize: '14px' }}>{criteria.value}</div>
-                                )}
-                            </WorkflowText>
-                        </SplitInline>
-                    </WorkflowStepViewContainer>
-                );
-            })}
+                            <Button
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/WorkflowCriterias.tsx:85]: criteria cant be signed

**Review by Andreas-Tonnessen — APPROVED** [review]

File: src/apps/ScopeChangeRequest/Api/ScopeChange/getHistory.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { httpClient } from '../../../../Core/Client/Functions/HttpClient';
+import { LogEntry } from '../../Types/scopeChangeRequest';
+
+export async function getHistory(id: string): Promise<LogEntry[]> {
+    const { scopeChange } = httpClient();
+
+    return await scopeChange.fetch(`api/scope-change-requests/${id}/history`).then((x) => x.json());
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Api/ScopeChange/getHistory.ts:7]: x !== history

File: src/apps/ScopeChangeRequest/Components/DetailView/Components/OriginLink.tsx [hunk 3]
```diff
@@ -23,21 +23,27 @@ export const OriginLink = ({ type, id }: OriginLinkProps): JSX.Element => {
...
+                        {type} - {id}
+                    </Link>
+                );
             }
             case 'NCR': {
                 if (!id) return <div>Error query without id</div>;
                 return (
                     <Link onClick={() => onClickRedirectOrigin(id)}>
-                        {type}-{id}
+                        {' '}
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/DetailView/Components/OriginLink.tsx:37]: remove

File: src/apps/ScopeChangeRequest/Components/DetailView/Components/RequestDetailView/Double.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+const SplitScreen = styled.div`
+    display: flex;
+    flex-direction: row;
+    flex-basis: 0;
+    overflow: scroll;
+`;
+
+/**
+ //TODO:
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/DetailView/Components/RequestDetailView/Double.tsx:104]: remove

File: src/apps/ScopeChangeRequest/Components/DetailView/Components/RequestDetailView/Single.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                <Value>
+                    <HistoryList />
+                </Value>
+            </Section>
+        </div>
+    );
+};
+
+/**
+ //TODO:
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/DetailView/Components/RequestDetailView/Single.tsx:92]: remove

File: src/apps/ScopeChangeRequest/Components/Form/Origins/SearchDCN.tsx [hunk 6]
```diff
@@ -56,7 +56,11 @@ export const SearchDCN = ({ setOriginId, originId }: PCSLinkProps): JSX.Element
 
             <Inline>
                 <div
-                    style={{ width: '211px', borderBottom: '5px solid #6F6F6F', fontSize: '16px' }}
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/Origins/SearchDCN.tsx:59]: StyledComponent

File: src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestForm.tsx [hunk 7]
```diff
@@ -38,16 +36,13 @@ export const ScopeChangeRequestForm = ({
     closeScrim,
     setHasUnsavedChanges,
 }: ScopeChangeRequestFormProps): JSX.Element => {
-    const formData = useForm<ScopeChangeRequest>(scopeChangeRequestSchema);
+    const formData = useForm<ScopeChangeRequest>(scopeChangeRequestSchema, {
+        phase: 'IC phase',
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestForm.tsx:40]: Remove

File: src/apps/ScopeChangeRequest/Components/MenuButton/Components/MenuButton.tsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                {buttonText} <Icon name="chevron_down" />
+            </Button>
+
+            <Menu
+                id="menu-complex"
+                aria-labelledby="anchor-complex"
+                open={showMenu}
+                anchorEl={anchorRef.current}
+                onClose={closeMenu}
+                // focus={focus}
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/MenuButton/Components/MenuButton.tsx:42]: remove

File: src/apps/ScopeChangeRequest/Components/SearchableDropdown/PCSLink.tsx [hunk 9]
```diff
@@ -207,30 +218,35 @@ function getIcon(x: TypedSelectOption): JSX.Element | null {
     if (x.type === 'tag') {
         return <Icon name="tag" color={tokens.colors.interactive.primary__resting.hex} />;
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/SearchableDropdown/PCSLink.tsx:219]: switch

File: src/apps/ScopeChangeRequest/Components/SearchableDropdown/PCSLink.tsx [hunk 10]
```diff
@@ -207,30 +218,35 @@ function getIcon(x: TypedSelectOption): JSX.Element | null {
...
+    border-bottom: none;
+`;
+
+const SearchContainer = styled.div`
+    flex-basis: 70%;
+    border-bottom: none;
 `;
 
-const Chip = styled.div`
-    text-align: center;
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/SearchableDropdown/PCSLink.tsx:221]: move to styles file

File: src/apps/ScopeChangeRequest/Components/Sidesheet/ScopeChangeSidesheet.tsx [hunk 11]
```diff
@@ -67,15 +71,25 @@ export const ScopeChangeSideSheet = (item: ScopeChangeRequest): JSX.Element => {
...
+                <Title>{data?.title}</Title>
                 <Button
                     variant="ghost_icon"
                     onClick={() => setEditMode(!editMode)}
                     disabled={!scopeChangeAccess.canPatch}
                 >
+                    <IconMenu
+                        items={[
+                            {
+                                label: 'Void request',
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Sidesheet/ScopeChangeSidesheet.tsx:88]: onClick

File: src/apps/ScopeChangeRequest/Components/Sidesheet/ScopeChangeSidesheet.tsx [hunk 12]
```diff
@@ -124,10 +137,14 @@ const TitleHeader = styled.div`
     align-items: center;
 `;
 
+const Divider = styled.div`
+    width: 0.3rem;
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Sidesheet/ScopeChangeSidesheet.tsx:141]: gap

File: src/apps/ScopeChangeRequest/Components/Workflow/Components/Contributors.tsx [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+    const contributorActions: MenuItem[] = [
+        {
+            label: ContributorActions.Confirm,
+            icon: <Icon name="check_circle_outlined" color="grey" />,
+            onClick: async () => await mutateAsync(),
+        },
+        {
+            label: ContributorActions.ConfirmWithComment,
+            icon: <Icon name="comment_add" color="grey" />,
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/Components/Contributors.tsx:40]: tokens.grey

File: src/apps/ScopeChangeRequest/Components/Workflow/Components/WorkflowCriterias.tsx [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+import { spawnConfirmationDialog } from '../../../../../Core/ConfirmationDialog/Functions/spawnConfirmationDialog';
+import { signCriteria } from '../../../Api/ScopeChange/Workflow';
+import { SignWithComment } from './SignWithComment';
+
+interface WorkflowCriteriasProps {
+    step: WorkflowStep;
+    criteria: Criteria;
+}
+
+export const WorkflowCriterias = ({ step, criteria }: WorkflowCriteriasProps): JSX.Element => {
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/Components/WorkflowCriterias.tsx:29]: rename

File: src/apps/ScopeChangeRequest/Components/Workflow/Components/WorkflowCriterias.tsx [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+
+    useEffect(() => {
+        if (person) {
+            reassignMutation({ type: 'RequireProcosysUserSignature', value: person.value });
+            setPerson(null);
+            setShowReassign(false);
+        }
+    }, [person]);
+
+    async function onSignStep() {
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/Components/WorkflowCriterias.tsx:43]: extract

File: src/apps/ScopeChangeRequest/Components/Workflow/Components/WorkflowCriterias.tsx [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+        setShowReassign(false);
+        setShowContributor(false);
+        setPerson(null);
+    };
+
+    interface ReassignArgs {
+        value: string;
+        type: 'RequireProcosysUserSignature';
+    }
+    const reassign = async ({ value, type }: ReassignArgs) => {
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/Components/WorkflowCriterias.tsx:100]: cleanup

File: src/apps/ScopeChangeRequest/Types/scopeChangeRequest.ts [hunk 17]
```diff
@@ -47,6 +72,8 @@ export interface ScopeChangeRequest extends ScopeChangeBaseModel {
     systems: System[];
     attachments: Attachment[];
     documents: Document[];
+    hasComments: boolean;
+    sequenceNumber: number;
     //workflow
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Types/scopeChangeRequest.ts:77]: Spør kjetil

**Pull request merged**